pallets/stablewap/src/lib.rs

```diff
@@ -0,0 +1,707 @@
+mod math;
+pub mod traits;
+pub mod types;
+pub mod weights;
+
+use crate::types::Balance;
+use weights::WeightInfo;
+
+#[cfg(test)]
+pub(crate) mod tests;
+
+#[cfg(any(feature = "runtime-benchmarks", test))]
+mod benchmarks;
+
+const POOL_IDENTIFIER: &str = "sts";
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::math::two_asset_pool_math::{
+		calculate_add_liquidity_shares, calculate_asset_b_reserve, calculate_in_given_out, calculate_out_given_in,
+		calculate_remove_liquidity_amounts,
+	};
+	use crate::traits::ShareAccountIdFor;
+	use crate::types::{AssetAmounts, Balance, PoolAssets, PoolId, PoolInfo};
+	use codec::HasCompact;
+	use core::ops::RangeInclusive;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use hydradx_traits::{Registry, ShareTokenRegistry};
+	use orml_traits::MultiCurrency;
+	use sp_runtime::traits::Zero;
+	use sp_runtime::ArithmeticError;
+	use sp_runtime::Permill;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(crate) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Identifier for the class of asset.
+		type AssetId: Member
+			+ Parameter
+			+ PartialOrd
+			+ Default
+			+ Copy
+			+ HasCompact
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen
+			+ TypeInfo;
+
+		/// Multi currency mechanism
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = Self::AssetId, Balance = Balance>;
+
+		/// Account ID constructor
+		type ShareAccountId: ShareAccountIdFor<PoolAssets<Self::AssetId>, AccountId = Self::AccountId>;
+
+		/// Asset registry mechnanism
+		type AssetRegistry: ShareTokenRegistry<Self::AssetId, Vec<u8>, Balance, DispatchError>;
+
+		/// The origin which can create a new pool
+		type CreatePoolOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+		/// Precision used in Newton's method to solve math equations iteratively.
+		#[pallet::constant]
+		type Precision: Get<Balance>;
+
+		/// Minimum pool liquidity
+		#[pallet::constant]
+		type MinimumLiquidity: Get<Balance>;
+
+		/// Amplification inclusive range. Pool's amp can be selected from the range only.
+		#[pallet::constant]
+		type AmplificationRange: Get<RangeInclusive<u32>>;
+
+		/// Minimum trading amount
+		#[pallet::constant]
+		type MinimumTradingLimit: Get<Balance>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	/// Existing pools
+	#[pallet::storage]
+	#[pallet::getter(fn pools)]
+	pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId<T::AssetId>, PoolInfo<T::AssetId>>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A pool was created.
+		PoolCreated {
+			id: PoolId<T::AssetId>,
+			assets: (T::AssetId, T::AssetId),
+			initial_liquidity: (Balance, Balance),
```

Comment:
I would leave this as is, because that is as it was provided in the parameters.  Otherwise, it introduces additional logic to order it.

Same applies for add liquidity event.

I'll think about it.

Reply:
leaving as is.